kernel/sync_block/src/lib.rs

```diff
@@ -89,18 +89,35 @@ impl MutexFlavor for Block {
             // Holder is either no longer running, or has released the lock.
             // Either way we will try the fast path one more time before moving
             // onto the slow path.
+
+            if let Some(guards) = Self::try_lock(mutex, data) {
+                return guards;
+            }
+
+            // Slow path
+            #[cfg(priority_inheritance)]
+            let deinherit_priority = scheduler::inherit_priority(&holder_task);
```

Comment:
This is a nice design for undoing the effects of priority inheritance. It'd be even nicer to use a guard type such that the effects of priority inheritance can also be undone in the case of exceptional execution (unwinding after a panic). Should be easy enough to wrap up the returned closure in a struct that implements `Drop` too.